neard/build.rs

```diff
@@ -45,16 +76,28 @@ fn get_git_version() -> Result<String> {
         let path = git_dir.join(subpath).canonicalize()?;
         println!("cargo:rerun-if-changed={}", path.display());
     }
-    let out = command("git", &["describe", "--always", "--dirty=-modified"]);
-    match out.as_ref().map(|ver| String::from_utf8_lossy(&ver)) {
-        Ok(std::borrow::Cow::Borrowed(version)) => Ok(version.trim().to_string()),
-        Ok(std::borrow::Cow::Owned(version)) => Err(anyhow!("git: invalid output: {}", version)),
-        Err(msg) => {
+    command(std::ffi::OsStr::new("git"), &["describe", "--always", "--dirty=-modified"]).or_else(
+        |error| {
             println!("cargo:warning=unable to determine git version");
-            println!("cargo:warning={}", msg);
+            println!("cargo:warning={}", error);
             Ok("unknown".to_string())
-        }
-    }
+        },
+    )
+}
+
+/// Returns rustc version.
+fn get_rustc_version() -> Result<String> {
+    command(&env_or("RUSTC", "rustc"), &["-vV"])
+        .and_then(|output| Ok(rustc_version::version_meta_for(&output)?))
+        .map(|meta| {
+            let channel = match meta.channel {
+                rustc_version::Channel::Dev => " dev",
+                rustc_version::Channel::Nightly => " nightly",
+                rustc_version::Channel::Beta => " beta",
+                rustc_version::Channel::Stable => "",
+            };
+            format!("{}{}", meta.semver, channel)
+        })
```

Comment:
```suggestion
        .map(|meta| meta.semver.to_string())
```

Semver already includes channel info: `1.59.0-beta.1`